File: atcoder/abc151B_achieve_the_goal_.nim
# Vicfred
# https://atcoder.jp/contests/abc151/tasks/abc151_b
# binary search
import strutils, sequtils, math

var n, k, m: int
(n, k, m) = stdin.readLine.split.map parseInt

let a = stdin.readLine.split.map parseInt

let score = sum(a)

var ini = 0
var fin = k
var mid = (ini+fin) div 2

while ini < fin:
  mid = (ini+fin) div 2
  if (score+mid) >= n*m:
    fin = mid
  else:
    ini = mid+1

if (score+ini) >= m*n:
  echo ini
else:
  echo -1

File: atcoder/abc151B_achieve_the_goal.py
# unihernandez22
# https://atcoder.jp/contests/abc151/tasks/abc151_b
# implementation

n, k, m = map(int, input().split())
a = list(map(int, input().split()))

if m*n - sum(a) <= k:
    print(max(m*n-sum(a), 0))
else:
    print(-1)
